src/platform/unix/mod.rs

```diff
@@ -39,6 +39,16 @@ const TEMP_FILE_TEMPLATE: &'static str = "/sdcard/servo/ipc-channel-shared-memor
 #[cfg(not(target_os="android"))]
 const TEMP_FILE_TEMPLATE: &'static str = "/tmp/ipc-channel-shared-memory.XXXXXX";
 
+unsafe fn new_sockaddr_un(path: *const c_char) -> (sockaddr_un, usize) {
+    let mut sockaddr: sockaddr_un = mem::zeroed();
+    libc::strncpy(sockaddr.sun_path.as_mut_ptr(),
+                  path, sockaddr.sun_path.len() - 1);
+    sockaddr.sun_family = libc::AF_UNIX as u16;
+    let len = mem::size_of::<c_short>() + (libc::strlen(sockaddr.sun_path.as_ptr())
+                                           as usize);
```

Comment:
While this happens to work on Linux and FreeBSD, I don't think it's really generic... As I said, this would probably require some pointer arithmetic -- but I still think the *right* way to handle this is to just use `sizeof sockaddr_un`, like in the Linux man page example, rather than trying to determine the actually used size dynamically.